### PR TITLE
fix(order-sync): Fix generating delivery update actions

### LINKED ISSUE #229: [order-sync] Fix adding new parcel to delivery
### Description
There is an error in [order sync](https://github.com/sphereio/sphere-node-sdk/blob/master/src/coffee/sync/utils/order.coffee#L68) method which throws an error when adding a new delivery parcel to an old delivery and simultaneously adding new delivery which is placed before the old delivery in the delivery array.

**Old object**
```js
{
  orderNumber: 'peanutbutter jelly',
  shippingInfo: {
    deliveries: [{
      id: 'c65e0661-5db8-4183-bcbc-6a4551a26d5b',  // OLD DELIVERY
      createdAt: '2017-05-22T07:33:02.202Z',
      items: [{
        id: '4dc17170-30ad-4b95-9a83-1388b40f5a1e',
        quantity: 9001
      }],
      parcels: [{
        id: '25098a64-95f6-48dd-b7ea-095d8224cb8e',
        createdAt: '2017-05-22T07:33:02.202Z',
        trackingData: {
          trackingId: '447883009643',
          carrier: 'dhl',
          isReturn: false
        }
      }]
    }]
  }
}
```

**New object**
```js
{
  orderNumber: 'peanutbutter jelly',
  shippingInfo: {
    deliveries: [{
      id: '1', // NEW DELIVERY
      items: [{
        id: '4dc17170-30ad-4b95-9a83-1388b40f5a1e',
        quantity: 100
      }],
      parcels: [{
        id: '1',
        trackingData: {
          trackingId: '123456789',
          carrier: 'TEST',
          provider: 'provider 1',
          providerTransaction: 'provider transaction 1',
          isReturn: false
        },
        measurements: {
          lengthInMillimeter: 100,
          heightInMillimeter: 200,
          widthInMillimeter: 200,
          weightInGram: 500
        }
      }]
    }, {
      id: 'c65e0661-5db8-4183-bcbc-6a4551a26d5b', // OLD DELIVERY
      items: [{
        id: '4dc17170-30ad-4b95-9a83-1388b40f5a1e',
        quantity: 9001
      }],
      parcels: [{
        id: '222',
        trackingData: {
          trackingId: '123456789',
          carrier: 'DHL',
          provider: 'provider',
          providerTransaction: 'transaction provider',
          isReturn: false
        },
        measurements: {
          lengthInMillimeter: 100,
          heightInMillimeter: 200,
          widthInMillimeter: 200,
          weightInGram: 500
        }
      }, {
        id: '25098a64-95f6-48dd-b7ea-095d8224cb8e',
        createdAt: '2017-05-22T07:33:02.202Z',
        trackingData: {
          trackingId: '447883009643',
          carrier: 'dhl',
          isReturn: false
        }
      }]
    }]
  }
}
```

### Expected Behavior
It should build sync actions properly.

### Current Behavior
Getting an error message:
```sh
'TypeError: Cannot read property \'id\' of undefined
    at /Users/xjunajan/dev/commercetools/orders-update/node_modules/sphere-node-sdk/lib/sync/utils/order.js:79:71
```

### PR DESCRIPTION
**Pull request by junajan**
#### Summary
Fixes #229 
When adding a new parcel to a delivery, the script goes through all deliveries to find proper `deliveryId`. It also ignores all diffs items starting with an underscore `_` because those items were removed and we do not currently support removed deliveries/parcels.

This PR also improves code coverage:
- [x] should generate correct update actions when a new delivery is added to the new object
- [x] should be agnostic of the index(position) of the deliveries in both new and old order object
- [x] should generate correction update actions a new parcel has been added
- [x] should work when no deliveries was existing in old order
- [x] should work when only new deliveries are provided without providing existing deliveries in the old order object (this is possible for orders because it's *currently* not possible to remove deliveries from an order)
- [ ] should throw if there is no deliveryId in the new order object (check if consistent with similar implementations in other entities)
> Here I discovered that the `deliveryId` has to be always present when adding new parcel because if it is missing, the script will add whole delivery instead of just a parcel.
- [x] should be idempotent, passing the same object should return empty actions

### CODE REVIEW

**Comment by junajan** [comment]
This "coveralls thing" is getting more and more painful :-I

**Comment by hisabimbola** [comment]
Nitpick: I think you meant to write `order-sync` not `category-sync` in the commit message

**Comment by junajan** [comment]
@hisabimbola I used amend to change it.

File: src/spec/sync/utils/order.spec.coffee [hunk 1]
```diff
@@ -237,23 +237,21 @@ describe 'OrderUtils', ->
...
   describe ':: actionsMapDeliveries', ->
 
-    it 'should return required actions for syncing deliveries', ->
-
+    it 'should return addDelivery action when syncing deliveries', ->
       orderChanged = _.deepClone ORDER
 
       # empty deliveries list
-      @order.shippingInfo.deliveries = []
+      delete @order.shippingInfo.deliveries
```
Comment (hisabimbola) [inline src/spec/sync/utils/order.spec.coffee]: let's set this to empty array, the API always return it as an empty array.

Comment (junajan) [inline src/spec/sync/utils/order.spec.coffee]: @hisabimbola updated here b34b31e

File: src/coffee/sync/utils/order.coffee [hunk 2]
```diff
@@ -59,13 +62,20 @@ class OrderUtils extends BaseUtils
         else
           # iterate over parcel instances
           _.chain deliveryDiff.parcels
-            .filter (item, key) -> key isnt '_t' and  _.isArray item
+            # filter out keys starting with '_' (old removed items or _t key)
+            .filter (item, key) -> key[0] isnt '_' and  _.isArray item
```
Comment (hisabimbola) [inline src/coffee/sync/utils/order.coffee]: why are we doing this now?

Can you add it as a comment also

Comment (junajan) [inline src/coffee/sync/utils/order.coffee]: Comment added here 9c1d300

**Pull request merged**